code/game/machinery/computer/medical.dm

```diff
@@ -228,25 +208,29 @@
 					sortBy = href_list["sort"]
 					order = initial(order)
 		else if(href_list["login"])
-			if(issilicon(usr))
-				src.active1 = null
-				src.active2 = null
-				src.authenticated = 1
-				src.rank = "AI"
-				src.screen = 1
-			else if(IsAdminGhost(usr))
-				src.active1 = null
-				src.active2 = null
-				src.authenticated = 1
-				src.rank = "Central Command"
-				src.screen = 1
-			else if(istype(src.scan, /obj/item/card/id))
-				src.active1 = null
-				src.active2 = null
-				if(src.check_access(src.scan))
-					src.authenticated = src.scan.registered_name
-					src.rank = src.scan.assignment
-					src.screen = 1
+			var/mob/M = usr
+			var/obj/item/card/id/I = M.get_idcard(TRUE)
```

Comment:
On bee silicons have an internal ID card so you can probably skip the silicone specific checks

Reply:
still needs different rank and assignment, so I'll leave it in